internal/hud/renderer.go

```diff
@@ -1,25 +1,60 @@
 package hud
 
 import (
-	"context"
+	"fmt"
+	"os"
 
+	"github.com/windmilleng/tcell"
 	"github.com/windmilleng/tilt/internal/hud/view"
 )
 
 type Renderer struct {
-	ttyPath string
-	ctx     context.Context
+	screen tcell.Screen
 }
 
 func (r *Renderer) Render(v view.View) error {
-	// TODO: draw v
-	// if r.ttyPath != "" {
-	//     log.Infof("drawing to tty: %s\n", r.ttyPath)
-	// }
+	if r.screen != nil {
```

Comment:
i worry about the race conditions here, it might be better to have a readyCh that closes when the screen is populated, and wait on readyCh here

Reply:
I was mostly relying on the `screen` assignment at the end of `SetUp` being atomic, but:
1. Apparently it's not fair to assume pointer assignments are atomic in go
2. You're right that that's asking for bugs

I've gone with a mutex instead of a channel because we don't want this method to block when no hud is attached.

Reply:
:+1: